### PR TITLE
SyncLock: Fix issues with single-playing syncables.

### PR DESCRIPTION
**Pull request by ywwg**
Discovered while investigating https://mixxx.discourse.group/t/vestax-tr-1-weird-behaviour-quantizing-drops-tempo/22715/7
When only one syncable is playing, use its beat distance to update the leader, so that seeks are properly handled

### CODE REVIEW

**Comment by ywwg** [comment]
I was unable to write a test to cover these cases :(

**Comment by Holzhaus** [comment]
> I was unable to write a test to cover these cases :(

Why? What was the issue?

**Comment by ywwg** [comment]
> Why? What was the issue?

I tried to create the same set of circumstances but the test always passed -- I couldn't get it to fail in the test environment.

**Comment by Holzhaus** [comment]
The likely cause, PR #3772, was merged into `2.3` but this PR is for `main`. Does the bug only occur in 2.4, not 2.3?

**Comment by Holzhaus** [comment]
Race condition ;-) you answered my question by opening #4156.

**Comment by daschuer** [comment]
How about merging https://github.com/mixxxdj/mixxx/pull/4156 to 2.3 first, merge 2.3 to main and rebase this on top of it to get rid of two commits doing the same?

**Comment by ywwg** [comment]
yes that's a good idea

**Comment by ywwg** [comment]
main has the other PR, so this one is ready to go too

File: src/engine/sync/enginesync.cpp [hunk 1]
```diff
@@ -371,7 +371,7 @@ void EngineSync::notifyPlayingAudible(Syncable* pSyncable, bool playingAudible)
             // Even if we didn't change leader, if there is only one player (us), then we should
```
Comment (uklotzde) [inline src/engine/sync/enginesync.cpp:371]: The comment doesn't match the new method invocation.

Comment (ywwg) [reply src/engine/sync/enginesync.cpp:371]: fixed

**Comment by daschuer** [comment]
Unfortunately #4156 does not completely fix the issue in 2.3 see my comments there. The same issue is already fixed here, unfortunately here it is shifted to another point. 

* Play two tracks in sync.
* Both quantize enabled  
* Nudge the follower track 
* Disable sync at the follower track 
* The follower is shifted into sync loosing the user offset. 
 The expected behavior is that the track continues to play without any tempo dip or shift.

**Review by daschuer — CHANGES_REQUESTED** [review]
There is a pending issue.

**Comment by ywwg** [comment]
hm ok I'll take a look

**Comment by Holzhaus** [comment]
Might be the same issue that @daschuer  mentioned, but I'm not sure: https://mixxx.discourse.group/t/vestax-tr-1-weird-behaviour-quantizing-drops-tempo/22715/14?u=hlzhs

**Comment by ywwg** [comment]
that feels like a bad beatgrid issue.  I'm looking in to daniel's issue now.

**Comment by ywwg** [comment]
> The follower is shifted into sync loosing the user offset.

Ah I see what is going on. This is a separate issue, I'd prefer to address it in a separate PR.  (Otherwise this becomes "let's fix absolutely every issue with sync lock in one PR" which is not reasonable)

**Comment by ywwg** [comment]
#4169 for the other issue

**Comment by ywwg** [comment]
anything more needed here?

**Comment by daschuer** [comment]
There is still an issue, when the explicit master is paused, quantize play doe not longer work, see: 
https://user-images.githubusercontent.com/1777442/128471887-988f96e5-675c-4701-baf8-f82901bda319.mp4

There is also an issue that the double detection is applied twice when repeated clicking on sync twice or using the right click: 
https://user-images.githubusercontent.com/1777442/128473395-b0149309-3949-4f27-8923-97d4e92a916a.mp4

Can you also take a look at the issue https://bugs.launchpad.net/mixxx/+bug/1939103 to backport the fix in this PR completely?

**Comment by ywwg** [comment]
do those issues only affect this branch, or do they affect main as well?  I have three outstanding synclock PRs and each fixes a different issue.  If those issues are new after these fixes, then they are bugs here that should be addressed.  If they also exist in main then they are out of scope.  These things will never get fixed if we allow every sync lock PR to expand to encompass all synclock bugs.

**Comment by daschuer** [comment]
> If they also exist in main then they are out of scope.

Yes of cause. I hope I will find time to check that soon with main. If you find time before to test it, it would be a good step forward.

**Comment by daschuer** [comment]
I can confirm that the first issue happens with plain master as well. I have filed a bug: 
https://bugs.launchpad.net/mixxx/+bug/1946677

**Comment by daschuer** [comment]
I no longer reproduce the second issue with this branch and with main. 
Hopfully it has been fixed in the meanwhile.

**Review by daschuer — APPROVED** [review]
LGTM, Thank you,

**Comment by daschuer** [comment]
Ups, I have merged this with a broken test. I will take a look.

**Pull request merged**